test/e2e/storage/testsuites/volumelimits.go

```diff
@@ -349,9 +349,9 @@ func getCSINodeLimits(cs clientset.Interface, config *PerTestConfig, nodeName st
 			return false, nil
 		}
 		var csiDriver *storagev1.CSINodeDriver
-		for _, c := range csiNode.Spec.Drivers {
+		for i, c := range csiNode.Spec.Drivers {
 			if c.Name == driverInfo.Name || c.Name == config.GetUniqueDriverName() {
-				csiDriver = &c
+				csiDriver = &csiNode.Spec.Drivers[i]
```

Comment:
how did this test work earlier ? It seems like the test was always returning the wrong limits unless the csDriver being looked at was first one ? Should the test be modified to catch this properly ?